registry/quilt_server/views.py

```diff
@@ -1556,21 +1570,26 @@ def profile():
     )
 
 @app.route('/api/payments/update_plan', methods=['POST'])
-@api()
+@api(enabled=HAVE_PAYMENTS)
 @as_json
 def payments_update_plan():
-    if not HAVE_PAYMENTS:
-        raise ApiException(requests.codes.not_found, "Payments not enabled")
-
     plan = request.values.get('plan')
     try:
         plan = PaymentPlan(plan)
     except ValueError:
         raise ApiException(requests.codes.bad_request, "Invalid plan: %r" % plan)
 
-    if plan not in (PaymentPlan.FREE, PaymentPlan.INDIVIDUAL, PaymentPlan.BUSINESS_ADMIN):
-        # Cannot switch to the BUSINESS_MEMBER plan manually.
-        raise ApiException(requests.codes.forbidden, "Not allowed to switch to plan: %r" % plan)
+    if TEAM_ID:
+        if not g.auth.is_admin:
+            raise ApiException(requests.codes.forbidden, "Only the admin can update plans")
+        # Can only switch to TEAM (from TEAM_UNPAID)
+        # if plan != PaymentPlan.TEAM:
+        if plan not in (PaymentPlan.TEAM, PaymentPlan.TEAM_UNPAID):
+            raise ApiException(requests.codes.forbidden, "Can only switch between team plans")
+    else:
+        if plan not in (PaymentPlan.FREE, PaymentPlan.INDIVIDUAL, PaymentPlan.BUSINESS_ADMIN):
```

Comment:
I have dropped support from `BUSINESS` plans completely in the frontend (and don't expect to need them).
If requires a migration to drop then I guess we can do that in a clean PR.

Reply:
Doesn't really require a migration (it's not like we have any actual business customers) - but I'll do it in a separate PR just to keep things simpler.

Also, technically, we should update the frontend in prod before we update the backend; it's not good to have the frontend try to use features that are no longer there.